src/osx/cocoa/window.mm

```diff
@@ -2568,6 +2574,63 @@ void wxOSXCocoaClassAddWXMethods(Class c, wxOSXSkipOverrides skipFlags)
     wxCocoaGestures::EraseForObject(this);
 }
 
+struct wxWidgetCocoaKeyDownEvent
+{
+    NSEvent* event;
+    int wxsent;
+};
+
+void wxWidgetCocoaImpl::BeginNativeKeyDownEvent( NSEvent* event )
+{
+    m_lastKeyDownEvent = event;
+    m_lastKeyDownWXSent = 0;
+}
+
+void wxWidgetCocoaImpl::EndNativeKeyDownEvent()
+{
+    m_lastKeyDownEvent = nil;
+    m_lastKeyDownWXSent = 0;
+}
+
+void wxWidgetCocoaImpl::PauseNativeKeyDownEvent(wxWidgetCocoaKeyDownEvent* &token)
+{
+    token = new wxWidgetCocoaKeyDownEvent;
+    
+    token->event = m_lastKeyDownEvent;
+    token->wxsent = m_lastKeyDownWXSent;
+    
+    m_lastKeyDownEvent = nil;
+}
+
+void wxWidgetCocoaImpl::ResumeNativeKeyDownEvent(wxWidgetCocoaKeyDownEvent* token)
+{
+    m_lastKeyDownEvent = token->event;
+    m_lastKeyDownWXSent = token->wxsent;
+    
+    delete token;
+}
+
+bool wxWidgetCocoaImpl::IsInNativeKeyDown()
+{
+    return m_lastKeyDownEvent != nil;
+}
+
+NSEvent* wxWidgetCocoaImpl::GetLastNativeKeyDownEvent()
+{
+    return m_lastKeyDownEvent;
+}
+
+void wxWidgetCocoaImpl::SetKeyDownSent()
+{
+    wxASSERT( m_lastKeyDownWXSent == 0 );
+    ++m_lastKeyDownWXSent;
```

Comment:
Why do we use `++` here and why is this an `int` and not a `bool` in the first place? It seems to only be possible to have `0` or `1` values for it, or am I missing something?

Reply:
I first had intended to have more versatility there, so the int was useful, but for the current usage and API a bool is indeed all we need